src/winshellext/F3DThumbnailProvider.cxx

```diff
@@ -151,42 +153,110 @@ IFACEMETHODIMP F3DThumbnailProvider::Initialize(LPCWSTR pszFilePath, DWORD)
 // Generate the thumbnail bitmap for the requested file.
 IFACEMETHODIMP F3DThumbnailProvider::GetThumbnail(UINT cx, HBITMAP* phbmp, WTS_ALPHATYPE* pdwAlpha)
 {
-  HRESULT ret = E_FAIL;
-  wchar_t warg[MAX_PATH * 2];
-  swprintf_s(warg, MAX_PATH * 2,
-    L"f3d --dry-run --quiet -sta --no-background --resolution=%d,%d \"%s\"",
-    cx, cx, m_filePath);
-  int argc;
-  char** argv;
-  ::ConvertWCommandLineToArgs(warg, argc, argv);
-
-  try
+  // Get a temporary PNG image file name
+  wchar_t lpTempPathBuffer[MAX_PATH];
+  wchar_t image_filename[MAX_PATH];
+
+  DWORD dwRetVal = GetTempPathW(MAX_PATH, lpTempPathBuffer);
+  if (dwRetVal > MAX_PATH || (dwRetVal == 0))
   {
-#if NDEBUG
-    vtkObject::GlobalWarningDisplayOff();
-#endif
+    return HRESULT_FROM_WIN32(GetLastError());
+  }
 
-    F3DLoader loader;
-    vtkNew<vtkImageData> image;
-    int r = loader.Start(argc, argv, image);
-    UINT imgWidth = static_cast<UINT>(image->GetDimensions()[0]);
+  if (GetTempFileName(lpTempPathBuffer, L"f3d", 0, image_filename) == 0)
+  {
+    return HRESULT_FROM_WIN32(GetLastError());
+  }
 
-    if (imgWidth == cx && r == EXIT_SUCCESS)
-    {
-      ::CreateBitmapFromBuffer(image, phbmp);
-      *pdwAlpha = WTSAT_ARGB;
-      ret = NOERROR;
-    }
+  wcscat_s(image_filename, L".png");
+
+  // Create command to run
+  wchar_t command[MAX_PATH * 3 + 20];
+  swprintf_s(command, MAX_PATH * 3 + 20,
+    L"\"%s\" --input \"%s\" --output \"%s\" --dry-run -sta --no-background --resolution %d,%d\"",
```

Comment:
probably need --quiet

Reply:
It was forgot in previous handler but yes, let's add it.